src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -34,12 +38,20 @@ public synchronized void setCurrentCommand(DriverCommand commandList) {
 	 */
 	public synchronized void setCurrentCommand(List<DriverCommand> commandList, String name) {
 		setCurrentCommand(new ICompoundCommand() {
+			Logger logger = Logger.getLogger("global");
 
 			List<DriverCommand> driverCommands = commandList;
 
 			@Override
 			public void execute(Job2dDriver driver) {
-				driverCommands.forEach((c) -> c.execute(driver));
+
+				driverCommands.forEach((c) -> {
+					c.execute(driver);
+					c.accept(commandVisitor);
+				});
+
+				this.logger.info("Wykonano łącznie " + commandVisitor.sumAll() + " operacji");
```

Comment:
do not modify this class. You should add test in menu or add observer showing logging command statistics when command is changed.